src/main/java/seedu/address/logic/commands/commandAssign/AssignTeacherToCourseCommand.java

```diff
@@ -35,18 +42,54 @@ public static boolean isValidDescriptor(AssignDescriptor assignDescriptor) {
     }
 
     @Override
-    public CommandResult execute(Model model) throws CommandException {
-        List<Course> lastShownList = model.getFilteredCourseList();
-        // TODO: Override equals by checking ID only then no need for this "stupid" for loop
-        for (Course course : lastShownList) {
-            if (course.getId().value.equals(this.assignDescriptor.getAssignID(PREFIX_COURSEID).value)) {
-                Course targetCourse = course;
-                targetCourse.setTeacherID(this.assignDescriptor.getAssignID(PREFIX_TEACHERID));
-                model.setCourse(targetCourse, targetCourse);
-                model.updateFilteredCourseList(PREDICATE_SHOW_ALL_COURSES);
-                return new CommandResult(String.format(MESSAGE_ASSIGNMENT_SUCCESS));
+    public CommandResult execute(Model model) throws CommandException, ParseException {
+        String courseidString = this.assignDescriptor.getAssignID(PREFIX_COURSEID).value;
+        String teacheridString = this.assignDescriptor.getAssignID(PREFIX_TEACHERID).value;
+        String courseName = "";
+        String teacherName = "";
+
+        boolean courseExists = false;
+        boolean teacherExists = false;
+        Course foundCourse = null;
+        Teacher foundTeacher = null;
+
+        for (Course course : model.getFilteredCourseList()) {
```

Comment:
Is there a function for has(Course x) or has(Teacher y), ideally we should be using slap instead of this whole chunk of code

Reply:
That will be a better implementation